### PR TITLE
feat(haqq-network, islamic-website): add robots.txt & sitemap.xml

### PR DESCRIPTION
**Pull request by hadzhehsen**

### CODE REVIEW

File: apps/haqq-website/app/sitemap.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  '/wallet',
+];
+
+export default async function sitemap(): Promise<MetadataRoute.Sitemap> {
+  const { posts } = await getBlogPosts();
+
+  const blogPosts = posts.map((post) => {
+    return {
+      url: `${DEPLOY_URL}/blog/${post.slug}`,
+      lastModified: new Date(post.date)?.toUTCString(),
```
Comment (olegshilov) [inline apps/haqq-website/app/sitemap.ts:26]: сюда бы надо дату изменения и не дату создания поста

File: apps/haqq-website/app/sitemap.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    };
+  });
+
+  const staticRoutes = staticUrls.map((el) => {
```
Comment (olegshilov) [inline apps/haqq-website/app/sitemap.ts]: not `el`. name it properly

File: apps/haqq-website/app/sitemap.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
+import { MetadataRoute } from 'next';
+import { DEPLOY_URL } from '../constants';
+import { getBlogPosts } from '../utils/get-blog-posts';
+
+const staticUrls = [
+  '',
```
Comment (olegshilov) [inline apps/haqq-website/app/sitemap.ts]:
```suggestion
  '/',
```

**Review by olegshilov — CHANGES_REQUESTED** [review]

File: apps/haqq-website/app/sitemap.ts (hunk 2 above)
Comment (hadzhehsen) [reply apps/haqq-website/app/sitemap.ts]: Done

File: apps/haqq-website/app/sitemap.ts (hunk 3 above)
Comment (hadzhehsen) [reply apps/haqq-website/app/sitemap.ts]: Done

File: apps/haqq-website/app/sitemap.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+const staticRoutes = [
+  '/',
+  '/audits',
+  '/blog',
+  '/brand-assets',
+  '/ecosystem',
+  '/ecosystem-fund',
+  '/events/scan',
+  '/events/sign-up',
```
Comment (olegshilov) [inline apps/haqq-website/app/sitemap.ts]: remove this routes

File: apps/islamic-website/app/robots.ts [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  const sitemapUrls = SUPPORTED_LOCALES.map((locale) => {
+    return new URL(`/${locale}/sitemap.xml`, SITE_URL).toString();
+  });
+
+  return {
+    rules: {
+      userAgent: '*',
+      allow: '/',
+    },
+    sitemap: sitemapUrls,
```
Comment (olegshilov) [inline apps/islamic-website/app/robots.ts]: here should be url to single sitemap, cuz you move it to `app` folder

File: apps/islamic-website/app/sitemap.ts [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+  '/values',
+  '/wallet',
+  '/whitepaper',
+];
+
+const academyModules = [
+  { moduleCount: 1, lessonsCount: 2, releaseDate: new Date('2023-12-12') },
+  { moduleCount: 2, lessonsCount: 3, releaseDate: new Date('2023-12-19') },
+  { moduleCount: 3, lessonsCount: 3, releaseDate: new Date('2023-12-26') },
+];
```
Comment (olegshilov) [inline apps/islamic-website/app/sitemap.ts]: use this array
https://github.com/haqq-network/frontend/blob/dev/libs/islamic-website/academy-page/src/lib/academy-preview-page/academy-preview-page.tsx#L24

File: apps/islamic-website/constants.ts [hunk 7]
```diff
@@ -1,8 +1,11 @@
-export const DEPLOY_URL = `https://${process.env['NEXT_PUBLIC_VERCEL_URL']}`;
+export const DEPLOY_URL = process.env['NEXT_PUBLIC_VERCEL_URL']
+  ? `https://${process.env['NEXT_PUBLIC_VERCEL_URL']}`
+  : 'http://localhost:4200';
```
Comment (olegshilov) [inline apps/islamic-website/constants.ts]:
```suggestion
export const DEPLOY_URL = process.env['VERCEL_URL']
  ? `https://${process.env['VERCEL_URL']}`
  : process.env['NEXT_PUBLIC_VERCEL_URL']
    ? `https://${process.env['NEXT_PUBLIC_VERCEL_URL']}`
    : 'http://localhost:4200'; ```

**Review by olegshilov — CHANGES_REQUESTED** [review]

File: apps/islamic-website/constants.ts (hunk 7 above)
Comment (hadzhehsen) [reply apps/islamic-website/constants.ts]: Done

File: apps/haqq-website/app/sitemap.ts (hunk 4 above)
Comment (hadzhehsen) [reply apps/haqq-website/app/sitemap.ts]: Done

File: apps/islamic-website/app/robots.ts (hunk 5 above)
Comment (hadzhehsen) [reply apps/islamic-website/app/robots.ts]: Done

**Review by olegshilov — APPROVED** [review]

**Pull request merged**